packages/react-core/src/components/Nav/NavItem.tsx

```diff
@@ -43,19 +45,27 @@ export const NavItem: React.FunctionComponent<NavItemProps> = ({
   ...props
 }: NavItemProps) => {
   const Component = component as any;
-
+  const { isNavOpen } = React.useContext(PageSidebarContext);
   const renderDefaultLink = (context: any): React.ReactNode => {
     const preventLinkDefault = preventDefault || !to;
     return (
-      <Component
-        href={to}
-        onClick={(e: any) => context.onSelect(e, groupId, itemId, to, preventLinkDefault, onClick)}
-        className={css(styles.navLink, isActive && styles.modifiers.current, className)}
-        aria-current={isActive ? 'page' : null}
-        {...props}
-      >
-        {children}
-      </Component>
+      <PageContextConsumer>
```

Comment:
don't need PageContextConsumer since you have the definitive answer to isNavOpen if you make the changes in my other comment :)